src/app/modules/verification/verification.service.ts

```diff
@@ -611,20 +614,33 @@ const checkSmsCountAndPerformAction = (
 }
 
 /**
- * Check whether the field in the form is verifiable.
- * If it is not, then we don't need to generate an OTP.
+ * Check whether we should generate an OTP according to the requirements:
+ * 1. the field in the form is verifiable, and
+ * 2. if the recipient is within the allowed countries set by the field
+ * If these conditions are not met, then don't generate an OTP.
  */
 export const shouldGenerateMobileOtp = (
   { form_fields }: Pick<IFormSchema, 'form_fields'>,
   fieldId: string,
+  recipient: string,
 ): ResultAsync<true, OtpRequestError> => {
```

Comment:
update the return type to also return `MalformedParametersError`?
Any reason why this error is used instead of reusing `OtpRequestError`? I would think all errors that occur in this function qualifies as a request error

Reply:
Seems that if anything goes wrong in those checks it's an issue with how the args of the call to `shouldGenerateMobileOtp` was formed, and its similar to how `MalformedParametersError` is used in `sendNewOtp`. Changing it to `OtpRequestError` is fine too, just that these errors are more of the type where "we can't find the info we need to send it" rather than "we are not going to send it" so i thought this was more appropriate. 

Noted on the updated return type tho - thanks for catching that.

Reply:
the second error is also `MalformedParametersError` because we need to access `allowIntlNumbers` which only exists in the mobile field

Reply:
cancel last, i'll change it to `OtpRequestError` :)